docs/source/upgrade/index.rst

```diff
@@ -82,13 +83,109 @@ has one *nonconsuming* choice that accepts a *Coin* contract, archives
+
+Let's see everything in action by following the steps to build and deploy version-1 of our coin. After this we'll see how to deploy and upgrade to version-2.
+
+First we'll need a sandbox ledger to which we can deploy.
+
+.. code-block:: none
+
+   $ daml sandbox --port 7001
+
+The version-1 project is named ``coinV1``. It contains just the original *Coin* definition template ``Coin``. Here is the ``daml.yaml`` setup
+
+.. literalinclude:: example/coinV1/daml.yaml
+  :language: yaml
+  :start-after: # BEGIN
+  :end-before: # END
+
+To make the example complete, we need a way to create some coins which have different parties for the issuer and owner. We take the standard approach of adding a :ref:`Propose-Accept workflow <intro propose accept>` to our example.
+
+.. literalinclude:: example/coinV1/daml/CoinV1.daml
+  :language: daml
+  :start-after: -- COIN_TRANSFER_BEGIN
+  :end-before: -- COIN_TRANSFER_END
+
+Now we can build and deploy version-1 of our Coin.
+
+.. code-block:: none
+
+   $ cd example/coinV1
+   $ daml build
+   $ daml deploy --port 7001
+
+
+Create some version-1 coins
+================================
+
+Let's create some coins. We'll use the navigator to connect to the ledger, and create two coins issued by Alice, and owned by Bob.
+
+.. code-block:: none
+
+   $ cd example/coinV1
+   $ daml navigator server localhost 7001
+
+We point a browser to ``localhost:4000``, and follow the steps:
+
+#. Login as Alice:
+    #. Create a coin with Alice as both issuer and owner.
+    #. Exercise the coin's *Transfer* choice, naming Bob as the new owner.
+    #. Create and transfer a 2nd coin in the same way.
+#. Login as Bob:
+    #. Exercise the *Accept* choice on both transfer contracts.
+
+
+Building and deploying version-2
+================================
+
+Now we setup the project for the improved coins containing the *amount* field. The new project is named ``coinV21`` and contains the templates for ``CoinWithAmount``, ``UpgradeCoinProposal`` and ``UpgradeCoinAgreement``. The project setup is:
+
+.. literalinclude:: example/coinV2/daml.yaml
+  :language: yaml
+  :start-after: # BEGIN
+  :end-before: # END
+
+The interesting thing about the project setup for ``coinV2`` is that ``coinV1-0.0.1.dar`` is listed as a dependency. This allows the upgrade templates to make reference to the ``Coin`` template from the ``CoinV1`` module, even though that module is part of original package already deployed to the ledger.
+
+The DAML for the upgrade contacts imports the modules for both the new and old coin versions.
+
+.. literalinclude:: example/coinV2/daml/UpgradeFromCoinV1.daml
+  :language: daml
+  :start-after: -- UPGRADE_MODULE_BEGIN
+  :end-before: -- UPGRADE_MODULE_END
+
+Now we can build and deploy version-2 of our Coin.
+
+.. code-block:: none
+
+   $ cd example/coinV2
+   $ daml build
+   $ daml deploy --port 7001
+
+
+Upgrade existing coins from version-1 to version-2
+================================
```

Comment:
```suggestion
==================================================
```
RST is great.

Reply:
Oh man. This is what caused my CI build to fail.. that's crazy!
How come I didn't get a local failure?

Reply:
The previous script does not treat warnings as fatal whereas the Bazel target does. If you run `bazel build //docs:docs` it will fail localyl as well. Rst is great :woman_facepalming: